### PR TITLE
feat(tests) add postgres to our e2e test framework

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

With this changes we are able to run test with postgres as a data
store

As a followup I will add more test for resiliency

### Testing

- [ ] Unit tests
- [x] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### CODE REVIEW

File: test/dockerfiles/Dockerfile.postgres [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  && mkdir /var/run/sshd \
+  && passwd -d root \
+  && chmod a+rwx /root
+
+RUN usermod -a -G sudo postgres \
+  && echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
+COPY $KUMA_ROOT/test/dockerfiles/scripts/postgres-up.sh postgres-up.sh
+
+RUN chmod +x postgres-up.sh
```
Comment (jpeach) [inline test/dockerfiles/Dockerfile.postgres:20]: If we chmod+x the script in git, do we still need to do it here?

File: mk/docker.mk [hunk 2]
```diff
@@ -89,7 +96,7 @@ images: images/release images/test
 images/release: image/kuma-cp image/kuma-dp image/kumactl image/kuma-init image/kuma-prometheus-sd ## Dev: Rebuild all Docker images
 
 .PHONY: images/test
-images/test: image/kuma-universal
+images/test: image/kuma-universal docker/build/postgres
```
Comment (jpeach) [inline mk/docker.mk:99]: Nit: might be preferable to add an empty `image/postgres` target just so this is consistent with the pattern of targets.

File: test/framework/interface.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ type deployOptions struct {
 	hdsDisabled      bool
 	serviceProbe     bool
 	isipv6           bool
+	storeType        string
```
Comment (jpeach) [inline test/framework/interface.go:41]: nit: can we make this a string type?
```Go
type DeploymentStoreType string
const StoreTypePostgres = DeploymentStoreType("postgres")

```

File: test/framework/universal_app.go [hunk 4]
```diff
@@ -193,36 +196,196 @@ type UniversalApp struct {
...
 	ip           string
 	verbose      bool
+	dockerOpts   *docker.RunOptions
 }
 
-func NewUniversalApp(t testing.TestingT, clusterName, dpName string, mode AppMode, isipv6, verbose bool, caps []string) (*UniversalApp, error) {
-	app := &UniversalApp{
-		t:            t,
-		ports:        map[string]string{},
+type UniversalAppOpts struct {
```
Comment (jpeach) [inline test/framework/universal_app.go:202]: I wonder if we really need to introduce this type. Could we just use functional options with `NewUniversalApp`, e.g.

```Go
app, err := NewUniversalApp(
      AppTestingOpt(t),
      AppVerboseOpt(true),
      AppEnableIPv6Opt(false),

```

File: test/framework/universal_app.go [hunk 5]
```diff
@@ -243,22 +406,18 @@ func NewUniversalApp(t testing.TestingT, clusterName, dpName string, mode AppMod
...
-func (s *UniversalApp) allocatePublicPortsFor(ports ...string) {
-	for _, port := range ports {
-		pubPortUInt32, err := util_net.PickTCPPort("", s.lastUsedPort+1, 11204)
-		if err != nil {
-			panic(err)
-		}
-		s.ports[port] = strconv.Itoa(int(pubPortUInt32))
-		s.lastUsedPort = pubPortUInt32
-	}
+func NewKumaCPApp() *UniversalAppOpts {
```
Comment (jpeach) [inline test/framework/universal_app.go:409]: This returns options, so maybe call it `NewKumaCPAppOpts`?

**Comment by bartsmykla** [comment]
Closing in favor of: https://github.com/kumahq/kuma/pull/2085

**Pull request closed without merging**